From a1892e0ed5c49c0632f2f0fd8b35aedd698b14a7 Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@redhat.com>
Date: Tue, 12 Aug 2014 11:30:57 -0400
Subject: [char] hwrng: virtio - re-arrange struct elements for better packing

Message-id: <3cc69fd813c7f4d52c62bd1ccb4b2d978c6f09ce.1407842211.git.amit.shah@redhat.com>
Patchwork-id: 87654
O-Subject: [RHEL7.1 kernel PATCH 7/9] virtio: rng: re-arrange struct elements for better packing
Bugzilla: 1119299
RH-Acked-by: Amos Kong <akong@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>

Re-arrange the elements of the virtrng_info struct to pack it better.

Signed-off-by: Amit Shah <amit.shah@redhat.com>
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
(cherry picked from commit 6062829fcdcae4518436f51336c0e05bd1f04806)
Signed-off-by: Amit Shah <amit.shah@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/char/hw_random/virtio-rng.c b/drivers/char/hw_random/virtio-rng.c
index 5799389..dc5d29d 100644
--- a/drivers/char/hw_random/virtio-rng.c
+++ b/drivers/char/hw_random/virtio-rng.c
@@ -31,11 +31,11 @@ static DEFINE_IDA(rng_index_ida);
 struct virtrng_info {
  struct hwrng hwrng;
  struct virtqueue *vq;
- unsigned int data_avail;
  struct completion have_data;
- bool busy;
  char name[25];
+ unsigned int data_avail;
  int index;
+ bool busy;
 };
 
 static bool probe_done;
-- 
1.7.1